scripts/test/MultiPlotting/Subplot_test.py

```diff
@@ -0,0 +1,150 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+import unittest
+
+from MultiPlotting.subplot.subPlot import subPlot
+from MultiPlotting.multiPlotting_context import PlottingContext
+
+from  Muon.GUI.Common import mock_widget
+try:
+    from unittest import mock
+except ImportError:
+    import mock
+
+
+def rm_logic(name):
+    if name == "two":
+       return False
+    return True
+
+class SubplotTest(unittest.TestCase):
+    def setUp(self):
+        self._qapp = mock_widget.mockQapp()
+        context = PlottingContext()
+        self.subplot = subPlot(context)
+        self.subplot.canvas.draw = mock.MagicMock()
+    
+
+    def setup_rm(self):
+        self.subplot._raise_rm_window = mock.Mock()
+        self.subplot._raise_selector_window = mock.Mock()
+        self.subplot._get_rm_window = mock.Mock()
+        self.subplot._createSelectWindow = mock.MagicMock()
+ 
+
+    def test_rmOnePlotNewWindow(self):
+        self.subplot._rm_window = None
+        self.subplot._selector_window = None
+
+        self.subplot._context.subplots["one"] = mock.Mock()
+        self.setup_rm()
+        self.subplot._rm()
+
+        self.assertEquals(self.subplot._raise_rm_window.call_count,0)
+        self.assertEquals(self.subplot._raise_selector_window.call_count,0)
+        self.assertEquals(self.subplot._get_rm_window.call_count,1)
+        self.assertEquals(self.subplot._createSelectWindow.call_count,0)
+
+    def test_rmOnePlotOldWindow(self):
+        self.subplot._rm_window = mock.Mock()
+        self.subplot._selector_window = None
+
+        self.subplot._context.subplots["one"] = mock.Mock()
+        self.setup_rm()
+        self.subplot._rm()
+
+        self.assertEquals(self.subplot._raise_rm_window.call_count,1)
+        self.assertEquals(self.subplot._raise_selector_window.call_count,0)
+        self.assertEquals(self.subplot._get_rm_window.call_count,0)
+        self.assertEquals(self.subplot._createSelectWindow.call_count,0)
+
+
```

Comment:
not sure whether this is a problem or not as flake8 is not run on test files but the python standard specifies a one line gap between functions in a class.